Bug: CI Workflow failing due to Docker update step.

### Describe the bug.

Our CI workflow is failing when we merge PRs, this occurs during the `Install Latest Docker Version` step. Error is caused by a dependency that wasn't removed/purged before installing the new docker version that comes packaged with that dependency.

Likely caused by a dependency update in new version, 2.20.0, of Docker Compose on 7-11-2023.

### Describe the steps to reproduce the behavior.

_No response_

### Expected behavior.

`The following packages have unmet dependencies:
 docker-ce : Depends: containerd.io (>= 1.6.4)
E: Unable to correct problems, you have held broken packages.
Error: Process completed with exit code 100.`

### Screenshots.

_No response_

### Operating Environment

N/A

### Additional Information

_No response_

--- a/.github/workflows/build-push.yml
+++ b/.github/workflows/build-push.yml
@@ -102,10 +102,12 @@ jobs:
       # Upgrade Docker engine version, needed for building images
       - name: Install Latest Docker Version
         run: |
+          sudo apt-get purge docker-ce docker-ce-cli containerd.io runc containerd moby-buildx moby-cli moby-compose moby-containerd moby-engine moby-runc
+
           curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
           sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu  $(lsb_release -cs)  stable"
           sudo apt-get update
-          sudo apt-get install docker-ce
+          sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
 
       # This is needed for our cross architecture docker image builds
       - name: Set up QEMU
@@ -373,10 +375,12 @@ jobs:
         # Upgrade Docker engine version, needed for building images
         - name: Install Latest Docker Version
           run: |
+            sudo apt-get purge docker-ce docker-ce-cli containerd.io runc containerd moby-buildx moby-cli moby-compose moby-containerd moby-engine moby-runc
+
             curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
             sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu  $(lsb_release -cs)  stable"
             sudo apt-get update
-            sudo apt-get install docker-ce
+            sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
 
         # This is needed for our cross architecture docker image builds
         - name: Set up QEMU

